### PR TITLE
Add documentation to API code

### PR DESCRIPTION
**Pull request by shawnrader**
Pull request for crate documentation for rust-fil-proofs-api.

### CODE REVIEW

**Review by vmx — COMMENTED** [review]
I haven't reviewed everything yet. But here are some general notes (besides the inline comments):

 - "storage miners" are called "storage providers" these days
 - Consistency (I don't care much about those, but I thought I mention them in case you care):
   - You sometimes start the explanation of an argument with uppercase letter sometimes not
   - Sometimes there's a dot at the end of the explanation of an argument, sometimes not.

File: src/update.rs [hunk 1]
```diff
@@ -38,7 +39,16 @@ fn empty_sector_update_encode_into_inner<Tree: 'static + MerkleTreeTrait<Hasher
...
+///
+/// Arguments:
+/// * `registered_proof` - Selected seal operation.
+/// * `new_replica_path` - Output path of new updated replica.
+/// * `new_cache_path` - Output path of new cache.
+/// * `sector_key_path` - Path to sector key originally used to seal sector.
+/// * `staged_data_path` - Path to staged data to encode into existing replica.
+/// * `piece_infos` - the piece info (commitment and byte length) for each piece in the sector.
+///
+/// Returns new commitments in EmptySectorUpdateEncoded struct.
```
[the remark this answers is not in the document]
Comment (shawnrader) [reply src/update.rs:51]: Thanks for the link tip, will make these changes.  Consistency in the arguments still needs some work.

File: src/update.rs [hunk 2]
```diff
@@ -162,6 +182,17 @@ fn empty_sector_update_remove_encoded_data_inner<
...
+/// # Arguments
+/// * `registered_proof` - Selected sector update proof.
+/// * `sector_key_path` - Path to sector key originally used to seal sector.
+/// * `sector_key_cache_path` - Path to write updated tree_r_last.
+/// * `replica_path` - File path of new sealed replica.
+/// * `replica_cache_path` - Directory cache path for replica (for p_aux).
+/// * `data_path` - File path for new staged data.
+/// * `comm_d_new` - Data commitment.
+///
+/// Returns [`Result`].
```
Comment (vmx) [inline src/update.rs:195]: As mentioned on a previous comment, `Result<()>` means that there is not return value.

File: src/seal.rs [hunk 3]
```diff
@@ -697,8 +697,7 @@ pub fn get_seal_inputs(
     )
 }
 
-// TODO: does this need to be public?
-pub fn get_seal_inputs_inner<Tree: 'static + MerkleTreeTrait>(
+fn get_seal_inputs_inner<Tree: 'static + MerkleTreeTrait>(
```
Comment (vmx) [inline src/seal.rs:700]: I'd probably do a separate PR for making all those non-public, as this is technically a breaking change and the rest of the doc updates are not.

File: src/post.rs [hunk 4]
```diff
@@ -1,3 +1,5 @@
+//! Generation and verification of Proof-of-Spacetime Merkle and zk-SNARK proofs.
```
Comment (vmx) [inline src/post.rs:1]: Should this be "Merkle trees"?

**Review by vmx — COMMENTED** [review]
Wow, that is really a lot of work and makes things so much better even if merged now! I've a few inline comments and then a list of very minor things that won't block merging:
 - It's `Spacetime`, hence the abbreviation "PoSt" is usually written with a lower case "t"
 - Sometimes you write "Merkle Tree" and sometimes with a lowercase "t"
 - You use two spaces to separate sentences. I know @cryptonemo is doing that, but I personally find it harder to read and it also is something that isn't used outside the US. I prefer using a style that is widely used internationally.
 - Putting identifiers like `comm_r` or `t_aux` in backticks makes them render nicer

File: src/seal.rs [hunk 5]
```diff
@@ -297,12 +302,37 @@ pub struct SealCommitPhase2Output {
...
 }
 
+/// Ensure that any persisted cached data for specified sector size is discarded.
+///
+/// # Arguments
+///
+/// * `sector_size` - Sector size associated with cache data to clear.
+/// * `cache_path` - Path to directory where cached data is stored.
+///
+/// Returns [`Result`]
```
[the remark this answers is not in the document]
Comment (shawnrader) [reply src/seal.rs:312]: I will just remove it since the return type is self explanatory

**Review by cryptonemo — APPROVED** [review]
